app/services/sso_service.rb

```diff
@@ -1,19 +1,27 @@
 # frozen_string_literal: true
 
-require 'saml/auth_fail_handler'
 require 'sentry_logging'
 
 class SSOService
   include SentryLogging
   include ActiveModel::Validations
   attr_reader :auth_error_code
```

Comment:
might need to add `:saml_response` here as an attr_reader method, or else line 112 of sessions_controller.rb will probably blow up... (Tests possibly passing because of that NoMethodError rescue block. 🤦‍♂️)

Reply:
There was a second `attr_reader` in this file with `saml_response` I'll tidy that.